fix: basic working rrf template

harish-aka-shivi opened the pull request:
- This PR adds a basic CRA custom-template with firebase and firestore added to the store instance.
- Change the basic layout similar to the output of generator-rrf output

harish-aka-shivi committed:
fix: basic working rrf template

harish-aka-shivi committed:
refactor (structure): change structure of code similar to generator output

prescottprue commented on template/src/containers/App/App.js:
```diff
@@ -1,36 +1,28 @@
         dispatch={store.dispatch}
         createFirestoreInstance={createFirestoreInstance}>
-        <Home />
+        <Router></Router>
```
This could be switched to `<Router />` since there isn't any children being passed. That said, we could just leave this as:

```jsx
<BrowserRouter>{routes}</BrowserRouter>
```

harish-aka-shivi replied:
The `<Router />` is doing some work that is creating Routes from `createRoutes` Function. That is why I separated it from `App` logic.  I will make it self closing tag. 
Open for suggestions

prescottprue commented on template/src/routes/Home/components/Home/Home.js:
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
-import logo from './logo.svg';
```
This original logo loading came straight from cra, is there a reason for referencing the file out of the statics folder instead? Is it because this was moved to a component?
We may want to call this `HomePage` instead of of just Home so that it is clear it is a route? Or maybe put it in a `routes` folder? Open to discussion on this

harish-aka-shivi replied:
The idea was to let the users know where the static files will be kept if they install and extend this template.
I agree with calling it `HomePage` instead of `Home`, so I will change it.

prescottprue commented on template/src/store/createStore.js:
```diff
@@ -0,0 +1,17 @@
+import { createStore, compose } from 'redux'
```
It doesn't seem like we are using compose right? We use the one from DEVTOOLS instead?

prescottprue commented on template/src/store/createStore.js:
```diff
@@ -0,0 +1,17 @@
+function configureStore(initialState, history) {
+  
+  const composeEnhancers = typeof window === 'object' &&
+  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
```
Hmm interesting, I haven't seen usage of `compose` from the window. Do we need this if we aren't adding any enhancers? I guess it is good to have in place for folks that want to use it, but maybe we should just use `compose` directly since it is already imported anyway?

harish-aka-shivi replied:
Yup, got it. What I can do is add the dev-tool enhancer using native `compose`. I am also adding an empty array for `middlewares`, in case the user wants to add it later, for convenience.

prescottprue commented on template/src/containers/App/Router.js:
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
```
We should stick to either single quotes or double quotes. Personally I prefer singles, but as long as they are all the same I am fine either way

harish-aka-shivi replied:
🤦, that's why we need prettier.

harish-aka-shivi committed:
fix(structure): fix the linting error and complex names

This commit:
- add the semicolon
- removes double quotes
- rename home component to HomePage

harish-aka-shivi committed:
refactor (store): change store to use redux compose function
